sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastNestedLoopJoinExec.scala

```diff
@@ -339,6 +340,18 @@ case class BroadcastNestedLoopJoinExec(
     )
   }
 
+  protected override def doPrepare(): Unit = {
+    if (!withinBroadcastThreshold && !sqlContext.conf.crossJoinEnabled) {
+      throw new AnalysisException(
+        s"""
+           |Both sides of this join are outside the broadcasting threshold and
```

Comment:
should it
- say a cross join is being considered (if it is)
- reference the broadcast threshold to allow the user to increase it?